galsim/base.py

```diff
@@ -628,13 +628,27 @@ def expand(self, scale):
         See magnify() for a version that applies a scale factor to the area while preserving surface
         brightness.
 
+        If `scale` is callable, it will be interpretted as a function of wavelength in nanometers.
+        The return value will then be a `ChromaticObject` with the given wavelength-dependent
+        scaling.
+
         @param scale    The factor by which to scale the linear dimension of the object.
 
         @returns the expanded object.
         """
-        new_obj = galsim.Transform(self, jac=[scale, 0., 0., scale])
+        if hasattr(scale, '__call__'):
+            def buildScaleJac(w):
+                s = scale(w)
+                return np.diag([s,s])
+            jac = buildScaleJac
+            if hasattr(self, 'noise'):
+                import warnings
+                warnings.warn("Cannot propagate noise through chromatic transformation.")
+        else:
+            jac = np.diag([scale, scale])
+        new_obj = galsim.Transform(self, jac=jac)
```

Comment:
> I'm trying to decide whether all these changes to the GSObject transformations are actually a net improvement.

I mostly just thought they'd be more convenient this way.  But you bring up some good points about potential proneness to bugs, for example.  I'll go ahead and revert this I think.